analyzer/library/composer/composer.go

```diff
@@ -19,21 +20,38 @@ func init() {
 
 const version = 1
 
-var requiredFiles = []string{"composer.lock"}
+var requiredFiles = []string{"/composer.lock", "/wp-includes/version.php"}
 
 type composerLibraryAnalyzer struct{}
 
 func (a composerLibraryAnalyzer) Analyze(target analyzer.AnalysisTarget) (*analyzer.AnalysisResult, error) {
-	res, err := library.Analyze(types.Composer, target.FilePath, target.Content, composer.Parse)
-	if err != nil {
-		return nil, xerrors.Errorf("error with composer.lock: %w", err)
+	parsedLibs := []godepTypes.Library{}
+	var err error
+	r := bytes.NewReader(target.Content)
+	if strings.HasSuffix(target.FilePath, "/composer.lock") {
+		parsedLibs, err = composer.Parse(r)
+		if err != nil {
+			return nil, xerrors.Errorf("failed to parse %s: %w", target.FilePath, err)
+		}
+	} else if strings.HasSuffix(target.FilePath, "/wp-includes/version.php") {
+		parsedLibs, err = composer.ParseWordPress(r)
+		if err != nil {
+			return nil, xerrors.Errorf("failed to parse %s: %w", target.FilePath, err)
+		}
 	}
-	return res, nil
+	if len(parsedLibs) == 0 {
```

Comment:
Other libraries return an AnalysisResult even if the detected library is empty.

Reply:
Sorry, no problems skip this process. 
https://github.com/aquasecurity/fanal/blob/2cd043a8486e276a20c5607814e2972e0cbb3db4/analyzer/analyzer.go#L98-L100